src/OpenTelemetry/Metrics/MeterProviderBuilderExtensions.cs

```diff
@@ -121,6 +121,53 @@ public static MeterProviderBuilder AddView(this MeterProviderBuilder meterProvid
             return meterProviderBuilder;
         }
 
+        /// <summary>
+        /// Sets the maximum number of Metric streams supported by the MeterProvider.
+        /// When no Views are configured, every instrument will result in one metric stream,
+        /// so this control the numbers of instruments supported.
+        /// When Views are configued, a single instrument can result in multiple metric stream,
+        /// so this controls the number of streams.
+        /// </summary>
+        /// <param name="meterProviderBuilder">MeterProviderBuilder instance.</param>
+        /// <param name="metricStreamLimit">The number of metric streams allowed.</param>
+        /// <returns>Returns <see cref="MeterProviderBuilder"/> for chaining.</returns>
+        /// <remarks>
+        /// If an instrument is created, but disposed later, this will still be contributing to the limit.
+        /// This may change in the future.
+        /// </remarks>
+        public static MeterProviderBuilder SetMetricStreamLimit(this MeterProviderBuilder meterProviderBuilder, int metricStreamLimit)
+        {
+            if (meterProviderBuilder is MeterProviderBuilderBase meterProviderBuilderBase)
+            {
+                meterProviderBuilderBase.SetMetricStreamLimit(metricStreamLimit);
+            }
+
+            return meterProviderBuilder;
+        }
+
+        /// <summary>
+        /// Sets the maximum number of MetricPoints allowed per metric stream.
```

Comment:
Bit of a nit, but this is the first time we're publicly introducing the term MetricPoint. So maybe

```suggestion
        /// Sets the maximum number of metric points allowed per metric stream.
```

I think the name of the method is fine `SetMetricPointPerMetricStreamLimit` ... unless there's a better term from the spec for describing the `MetricPoint` concept?

Reply:
Personal gut feeling - maybe for the user "cardinality" is something easier to digest?

Reply:
How about `SetTimeSeriesLimit`?

Reply:
maxTimeseries seems a good option.

Reply:
cardinality - i think this would be a better fit, if the limit was for limiting the cardinality of an individual tag.

Reply:
Check this out https://en.wikipedia.org/wiki/Cardinality

Reply:
lets me proceed as is now, to unblock release.
Will discuss the naming issues in next SIG meeting. We have ~10 more days before planned stable release, for coming up with better names)